src/components/Donate/DonateDialog.js

```diff
@@ -4,47 +4,83 @@ import "@ui5/webcomponents/dist/Dialog";
 import "@ui5/webcomponents/dist/MessageStrip";
 
 import "./DonateDialog.css";
+import { BLOOD_BANK_API } from './../../app-config'
 
 class DonateDialog extends Component {
 
   constructor(props) {
     super(props);
 
     this.state = {
-
+      selectedHospital: '',
+      hospitals: []
     };
 
     this.dialogRef = React.createRef();
+    this.comboboxRef = React.createRef();
     this.createButtonRef = React.createRef();
     this.cancelButtonRef = React.createRef();
 
     this.create = this.create.bind(this);
     this.close = this.close.bind(this);
+    this.onComboboxChange = this.onComboboxChange.bind(this);
   }
 
   componentDidMount() {
     if (this.dialogRef.current) {
       this.createButtonRef.current.addEventListener("click", this.create);
       this.cancelButtonRef.current.addEventListener("click", this.close);
     }
+
+    this.comboboxRef.current.addEventListener('change', this.onComboboxChange);
   }
 
   componentWillUnmount() {
     if (this.dialogRef.current) {
       this.createButtonRef.current.removeEventListener("click", this.create);
       this.cancelButtonRef.current.removeEventListener("click", this.close);
     }
+
+    if (this.comboboxRef.current) {
+      this.comboboxRef.current.removeEventListener('change', this.onComboboxChange);
+    }
   }
 
   open() {
     this.dialogRef.current.open();
   }
 
+  onComboboxChange() {
+    const selectedValue = this.comboboxRef.current.value;
+    if (selectedValue) {
+      this.setState({
+        selectedHospital: selectedValue
+      });
+    }
+  }
+
   close() {
     this.dialogRef.current.close();
   }
 
   create() {
+    const hospital = this.state.selectedHospital;
+    if (hospital) {
+      const url = BLOOD_BANK_API + '/donations/';
+      fetch(url, {
+        method: 'POST',
+        body: JSON.stringify({
+          "userID": localStorage.getItem("userID") || "",
+          "bloodcenter": hospital,
+          "status": "In Progress"
+        })
+      }).then(() => {
+        console.log("Your donation request sent successfully! Please refresh to see the latest status")
```

Comment:
Maybe refresh all data when created?

Also, 'date' field seems to be empty when creating new donation request.

Reply:
Fixed but I am not sure if it the 'React' way... please verify.